app/src/main/java/com/android/streetworkapp/ui/navigation/InfoDialogs.kt

```diff
@@ -0,0 +1,137 @@
+import com.android.streetworkapp.ui.utils.DialogType
+
+/**
+ * This class manages the InfoDialogs that are displayed when the user clicks on the info action. It
+ * is responsible for displaying the correct dialog for the current screen.
+ *
+ * @param showDialog - The display state of the InfoDialogs.
+ * @param screenName - The name of the current screen.
+ * @param topAppBarManager - The TopAppBarManager instance (used to set the action callback).
+ */
+class InfoDialogManager(
+    private val showDialog: MutableState<Boolean>,
+    private val screenName: MutableState<String?>,
+    private val topAppBarManager: TopAppBarManager?
+) {
+  private val screens = getScreens()
+
+  // Mapping Screen to InfoDialog
+  private val infoDialogs: MutableMap<String, InfoDialog> =
+      mutableMapOf(
+          "default" to defaultInfoDialog(),
+          screens.MAP to
+              InfoDialog(
+                  "Map",
+                  "How does the map work ?",
+                  "This is the map, here you can see each park that is near you. \n\n You can navigate to see more. Click on a park to see its details, such as their events !"),
+          screens.PROFILE to
+              InfoDialog(
+                  "Profile",
+                  "What is the Profile ?",
+                  "This is your profile page, you can see your friends and your score. \n\n You can also add friends using the 'Add friend' button and you can also access your settings by clicking the gear icon on the top right."),
+          screens.ADD_FRIEND to
+              InfoDialog(
+                  "AddFriend",
+                  "How do friends work ?",
+                  "This is where you can add a friend. \n\n To add, you need to have a friend nearby and activate bluetooth and location services. \n\n Once you are friends you will be able to do workouts with each other. \n\n Have fun together !"),
+          screens.PARK_OVERVIEW to
+              InfoDialog(
+                  "ParkOverview",
+                  "How does parks work ?",
+                  "This is the park overview, here you can see the park's details and the events that are planned in it. \n\n You can create an event or join an already existing one. You can also rate the park !"),
+          screens.ADD_EVENT to
+              InfoDialog(
+                  "AddEvent",
+                  "How to create an event ?",
+                  "This is your firsts steps towards creating an event ! \n\n You simply have to add a cool title, then, describe your event : What are you going to do there ? \n\n Finally, set the date and time and you are good to go !"),
+          screens.EVENT_OVERVIEW to
+              InfoDialog(
+                  "EventOverview",
+                  "What is the Event Overview ?",
+                  "This is the event overview, here you can access all of the information that you need. \n\n You are free to join the event if you want to participate !"),
```

Comment:
I do not completely agree because it is inside the Event Overview; the user must have clicked on a specific event. I think that it is better to specify the event in this place, whereas in parkoverview it would totally make sense.

Reply:
actually you are right, i misread and did not fully pay attention to the context in which this message was used
Apologies